Loop/Managers/Alerts/AlertManager.swift

```diff
@@ -224,6 +211,23 @@ extension AlertManager {
 // MARK: Extensions
 
 extension FileManager {
+    
+    fileprivate func ensureDirectoryExists(_ directory: URL?, log: DiagnosticLog) {
+        guard let directory = directory else {
+            return
+        }
+        do {
+            var isDirectory: ObjCBool = false
+            let exists = fileExists(atPath: directory.absoluteString, isDirectory: &isDirectory)
+            if !(exists && isDirectory.boolValue) {
```

Comment:
Why try calling createDirectory if the path exists and is a file?

Reply:
I figured it was more thorough.  In practice this shouldn't be a problem, but if it ever happened I wanted the failure to be clearer (because it wouldn't create a directory on top of a file)

Reply:
```